### PR TITLE
Added textInteractionEnabled prop

### PR DESCRIPTION
**Pull request by Arjan-Zuidema**
Added prop `textInteractionEnabled`. Only supported on iOS 15+

When set to `false` disables all text interaction.

### CODE REVIEW

**Comment by clemensmol** [comment]
Merging would be awesome, this is also a nice workaround for #2177.

**Review by amanBhawsar — APPROVED** [review]

File: apple/RNCWebView.m [hunk 1]
```diff
@@ -333,6 +333,10 @@ - (WKWebViewConfiguration *)setUpWkWebViewConfig
     [prefs setValue:@TRUE forKey:@"javaScriptCanOpenWindowsAutomatically"];
     _prefsUsed = YES;
   }
+  if (!_textInteractionEnabled) {
+    [prefs setValue:@FALSE forKey:@"textInteractionEnabled"];
+    _prefsUsed = YES;
+  }
```
Comment (TheAlmightyBob) [inline apple/RNCWebView.m:339]: I think this also needs the `#if` check around it?

And possibly also a runtime `if (@available` check?

**Review by TheAlmightyBob — CHANGES_REQUESTED** [review]
Please update `WebViewTypes` also.

And according to the [docs](https://developer.apple.com/documentation/webkit/wkpreferences/3727362-textinteractionenabled?changes=la_1&language=objc), it looks like this was actually introduced with iOS14.5, not iOS15?

File: apple/RNCWebView.m (hunk 1 above)
Comment (Arjan-Zuidema) [reply apple/RNCWebView.m]: Fixed this with both a #if and @availabel check for iOS 14.5

**Comment by Arjan-Zuidema** [comment]
> Please update `WebViewTypes` also.
> 
> And according to the [docs](https://developer.apple.com/documentation/webkit/wkpreferences/3727362-textinteractionenabled?changes=la_1&language=objc), it looks like this was actually introduced with iOS14.5, not iOS15?

Updated the `WebViewTypes`

File: apple/RNCWebViewManager.m [hunk 2]
```diff
@@ -104,6 +104,7 @@ - (RCTUIView *)view
 #endif
 
 #if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 150000 /* iOS 15 */
+RCT_EXPORT_VIEW_PROPERTY(textInteractionEnabled, BOOL)
```
Comment (TheAlmightyBob) [inline apple/RNCWebViewManager.m:107]: This is still gating it under iOS 15...?

File: docs/Reference.md [hunk 3]
```diff
@@ -1377,6 +1378,24 @@ Example:
 
 ---
 
+### `textInteractionEnabled`[⬆](#props-index)<!-- Link generated with jump2header -->
+
+If false indicates to WebKit that a WKWebView will not interact with text, thus not showing a text selection loop. Only applicable for iOS 15 or greater.
```
Comment (TheAlmightyBob) [inline docs/Reference.md:1383]: Possibly should update this to say 14.5

File: src/WebViewTypes.ts [hunk 4]
```diff
@@ -672,6 +673,15 @@ export interface IOSWebViewProps extends WebViewSharedProps {
    */
   limitsNavigationsToAppBoundDomains?: boolean;
 
+  /**
+   * If false indicates to WebKit that a WKWebView will not interact with text, thus
+   * not showing a text selection loop. Only applicable for iOS 15 or greater.
```
Comment (TheAlmightyBob) [inline src/WebViewTypes.ts:678]: Should probably update to say 14.5

**Review by TheAlmightyBob — CHANGES_REQUESTED** [review]
Looking pretty close!

It would also be good to test on a 14.5 simulator and make sure it really does still work as you expect there, if you haven't already

File: src/WebViewTypes.ts (hunk 4 above)
Comment (Arjan-Zuidema) [reply src/WebViewTypes.ts]: Updated to version 14.5

File: docs/Reference.md (hunk 3 above)
Comment (Arjan-Zuidema) [reply docs/Reference.md]: Updated to version 14.5

File: apple/RNCWebViewManager.m (hunk 2 above)
Comment (Arjan-Zuidema) [reply apple/RNCWebViewManager.m]: Created an iOS 14.5 section for this

**Comment by Arjan-Zuidema** [comment]
> Looking pretty close!
> 
> It would also be good to test on a 14.5 simulator and make sure it really does still work as you expect there, if you haven't already

Fixed the problems you pointed out, also tested on a simulator running iOS 14.5 this time :)

**Review by TheAlmightyBob — APPROVED** [review]

**Comment by TheAlmightyBob** [comment]
Thanks @Arjan-Zuidema!

**Pull request merged**